shopfloor/services/zone_picking.py

```diff
@@ -0,0 +1,1080 @@
+            message=message,
+        )
+
+    def _response_for_confirm_set_line_destination(
+        self, zone_location, picking_type, move_line, message=None
+    ):
+        return self._response(
+            next_state="confirm_set_line_destination",
+            data=self._data_for_move_line(zone_location, picking_type, move_line),
+            message=message,
+        )
+
+    def _response_for_zero_check(
+        self, zone_location, picking_type, location, message=None
+    ):
+        return self._response(
+            next_state="zero_check",
+            data=self._data_for_location(zone_location, picking_type, location),
+            message=message,
+        )
+
+    def _response_for_change_pack_lot(
+        self, zone_location, picking_type, move_line, message=None
+    ):
+        return self._response(
+            next_state="change_pack_lot",
+            data=self._data_for_move_line(zone_location, picking_type, move_line),
+            message=message,
+        )
+
+    def _response_for_unload_all(
+        self, zone_location, picking_type, move_lines, message=None
+    ):
+        return self._response(
+            next_state="unload_all",
+            data=self._data_for_move_lines(zone_location, picking_type, move_lines),
+            message=message,
+        )
+
+    def _response_for_confirm_unload_all(
+        self, zone_location, picking_type, move_lines, message=None
+    ):
+        return self._response(
+            next_state="confirm_unload_all",
+            data=self._data_for_move_lines(zone_location, picking_type, move_lines),
+            message=message,
+        )
+
+    def _response_for_unload_single(
+        self, zone_location, picking_type, move_line, message=None, popup=None
+    ):
+        return self._response(
+            next_state="unload_single",
+            data=self._data_for_move_line(zone_location, picking_type, move_line),
+            message=message,
+            popup=popup,
+        )
+
+    def _response_for_unload_set_destination(
+        self, zone_location, picking_type, move_line, message=None
+    ):
+        return self._response(
+            next_state="unload_set_destination",
+            data=self._data_for_move_line(zone_location, picking_type, move_line),
+            message=message,
+        )
+
+    def _response_for_confirm_unload_set_destination(
+        self, zone_location, picking_type, move_line
+    ):
+        return self._response(
+            next_state="confirm_unload_set_destination",
+            data=self._data_for_move_line(zone_location, picking_type, move_line),
+        )
+
+    def _data_for_select_picking_type(self, zone_location, picking_types):
+        return {
+            "zone_location": self.data.location(zone_location),
+            # available picking types to choose from
+            # TODO add lines count and priority lines count in the data
+            "picking_types": self.data.picking_types(picking_types),
```

Comment:
to agree on key names, please call the extra keys like this:

`lines_count`  # total lines count
`picking_count` # total operations/picking count
`priority_lines_count` # total lines count from priority pickings
`priority_picking_count` # total operations/picking count w/ high priority

Reply:
Added in https://github.com/camptocamp/wms/pull/33/commits/2924da9cdee470188a0347bca1e96bc82b771c4b
Not sure about how to compute these numbers exactly, but a first implementation is done.